packages/grid/x-data-grid/src/components/columnsManagement/GridColumnsManagement.tsx

```diff
@@ -0,0 +1,356 @@
+        if (col.hideable && (togglableColumns == null || togglableColumns.includes(col.field))) {
+          if (isVisible) {
+            // delete the key from the model instead of setting it to `true`
+            delete newModel[col.field];
+          } else {
+            newModel[col.field] = false;
+          }
+        }
+      });
+
+      return apiRef.current.setColumnVisibilityModel(newModel);
+    },
+    [apiRef, columns, getTogglableColumns],
+  );
+
+  const handleSearchValueChange = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setSearchValue(event.target.value);
+    },
+    [],
+  );
+
+  const currentColumns = React.useMemo(() => {
+    const togglableColumns = getTogglableColumns ? getTogglableColumns(sortedColumns) : null;
+
+    const togglableSortedColumns = togglableColumns
+      ? sortedColumns.filter(({ field }) => togglableColumns.includes(field))
+      : sortedColumns;
+
+    if (!searchValue) {
+      return togglableSortedColumns;
+    }
+
+    return togglableSortedColumns.filter((column) =>
+      searchPredicate(column, searchValue.toLowerCase()),
+    );
+  }, [sortedColumns, searchValue, searchPredicate, getTogglableColumns]);
+
+  const hideableColumns = React.useMemo(
+    () => currentColumns.filter((col) => col.hideable),
+    [currentColumns],
+  );
+
+  const allHideableColumnsVisible = React.useMemo(
+    () =>
+      hideableColumns.every(
+        (column) =>
+          columnVisibilityModel[column.field] == null ||
+          columnVisibilityModel[column.field] !== false,
+      ),
+    [columnVisibilityModel, hideableColumns],
+  );
+
+  const allHideableColumnsHidden = React.useMemo(
+    () => hideableColumns.every((column) => columnVisibilityModel[column.field] === false),
+    [columnVisibilityModel, hideableColumns],
+  );
+
+  const firstSwitchRef = React.useRef<HTMLInputElement>(null);
+
+  React.useEffect(() => {
+    if (autoFocusSearchField) {
+      searchInputRef.current!.focus();
+    } else if (firstSwitchRef.current && typeof firstSwitchRef.current.focus === 'function') {
+      firstSwitchRef.current.focus();
+    }
+  }, [autoFocusSearchField]);
+
+  let firstHideableColumnFound = false;
+  const isFirstHideableColumn = (column: GridColDef) => {
+    if (firstHideableColumnFound === false && column.hideable !== false) {
+      firstHideableColumnFound = true;
+      return true;
+    }
+    return false;
+  };
+
+  return (
+    <React.Fragment>
+      <GridColumnsManagementHeader className={classes.header} ownerState={rootProps}>
+        <rootProps.slots.baseTextField
+          placeholder={apiRef.current.getLocaleText('columnsManagementSearchTitle')}
+          inputRef={searchInputRef}
+          value={searchValue}
+          onChange={handleSearchValueChange}
+          variant="outlined"
+          size="small"
+          InputProps={{
+            startAdornment: <rootProps.slots.quickFilterIcon sx={{ pr: 1 }} />,
+          }}
```

Comment:
```suggestion
          InputProps={{
            startAdornment: (
              <rootProps.slots.baseInputAdornment position="start">
                <rootProps.slots.quickFilterIcon />
              </rootProps.slots.baseInputAdornment>
            ),
            sx: { pl: 1.5 },
          }}
```

To avoid shrinking the icon:

<img width="307" src="https://github.com/mui/mui-x/assets/13808724/4b382ffb-7dd4-44bb-88a1-32c442b98046">